source/contributed/typescript-and-screeps-1.md

```diff
@@ -0,0 +1,207 @@
+
+```bash
+$ cd my_proj/
+# start an npm project
+/my_proj$ npm init
+# start a typescript project (only generates the config)
+/my_proj$ tsc --init
+# initialize typings
+/my_proj$ typings init
+# if you are using tslint
+/my_proj$ tslint --init
+# initialize a git repo (you are using git, aren't you?)
+/my_proj$ git init
+
+```
+
+The main result of this is to generate default `tsconfig.json` and `typings.json` files.
+
+### tsconfig.json
+
+Yup.  We're *still* not ready to start coding yet (but almost!).  We need to let TypeScript know about our project and our target environment.
+
+To set up TypeScript for Screeps, we need to make sure the following are set as a bare-minimum in the `tsconfig.json`:
+
+```JSON
+{
+  "compilerOptions": {
+    "module": "commonjs",           /* Export NodeJS style modules (required for Screeps) */
+    "target": "es6",                /* Export as ES6 (so we get the cool features) */
+    "experimentalDecorators": true  /* We are going to use these later in this guide */
+  },
+  "include": [          /* Tell the compiler which directories contain your actual code */
+    "src/**/*.ts",      /* Tell TypeScript to include all our TypeScript files */
+    "src/**/*.d.ts",    /* Tell TypeScript to include all our definition files */
+    "typings/**/*.d.ts" /* Tell TypeScript about our installed Typings
+  ]
+}
+```
+
+### Typings
+
+The `typings` tool is how we install TypeScript typings for external JS modules (such as those from `npm`).
+
+Lets install the typings for Screeps, Lodash (screeps is using v3.10.1 at the time of this article), and Node (v6.10.3 at the time of this article).  This way the TypeScript compiler can know about these libraries, even though they aren't actually present in your code
+
+```bash
+# (don't worry, global in this context means "ambient" [ see below ]
+# it's not actually installing the typings globally on your computer)
+# Install Screeps Typings
+/my_proj$ typings install --save --global github:screepers/Screeps-Typescript-Declarations/dist/screeps.d.ts 
+# Install Node Typings from definately typed repo for version 6
+/my_proj$ typings install --save --global "dt~node@6.0.0"
+# Install Lodash Typings from definately typed repo for version 3.10
+/my_proj$ typings install --save --global "dt~lodash@3.10.0"
+```
+
+#### Newbie Gotcha: Ambient Declarations vs. Module Declarations
+
+As you may have guessed, typing is a key feature of TypeScript.  However, a common "gotcha" for beginners is the difference between "ambient" (also called "global") typing declarations, and "module" typing declarations.
+
+Above, we installed the typings as "global".  This means the declarations inside are just assumed to be available everywhere in your project, without having to `import` or `require` them.
+
+For example, `Creep` is ambient, because we can just call `Creep.moveTo(...)` without having to first call `import { Creep } from "some-module";`.
```

Comment:
> we can just call `Creep.moveTo(...)`
This is probably a bad example, as `Creep.moveTo` would be a static function. Maybe use `Game.creeps` instead.